### PR TITLE
add exercise_3

### PR DESCRIPTION
**Pull request by nguyenvanman**

### CODE REVIEW

File: Exercise_3/MainActivity.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    [Activity(Label = "Exercise_3", MainLauncher = true)]
+    public class MainActivity : Activity
+    {
+        [InjectView(Resource.Id.edFirstCofficient)] private EditText edFirstCofficient;
+        [InjectView(Resource.Id.edSecondCofficient)] private EditText edSecondCofficient;
+        [InjectView(Resource.Id.edThirdCofficient)] private EditText edThirdCofficient;
+        [InjectView(Resource.Id.tvResultTitle)] private TextView tvResultTitle;
+        [InjectView(Resource.Id.tvResult)] private TextView tvResult;
+        [InjectOnClick(Resource.Id.btnSolve)]
+        void Solve(object sender, EventArgs e)
```
Comment (voquanghoa) [inline Exercise_3/MainActivity.cs]: Missing modifier `private`

Comment (nguyenvanman) [reply Exercise_3/MainActivity.cs]: Done

File: Exercise_3/MainActivity.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    [Activity(Label = "Exercise_3", MainLauncher = true)]
+    public class MainActivity : Activity
+    {
+        [InjectView(Resource.Id.edFirstCofficient)] private EditText edFirstCofficient;
+        [InjectView(Resource.Id.edSecondCofficient)] private EditText edSecondCofficient;
+        [InjectView(Resource.Id.edThirdCofficient)] private EditText edThirdCofficient;
+        [InjectView(Resource.Id.tvResultTitle)] private TextView tvResultTitle;
+        [InjectView(Resource.Id.tvResult)] private TextView tvResult;
+        [InjectOnClick(Resource.Id.btnSolve)]
+        private void Solve(object sender, EventArgs e)
```
Comment (voquanghoa) [inline Exercise_3/MainActivity.cs:18]: The method is too long. Need to refactoring

Comment (nguyenvanman) [reply Exercise_3/MainActivity.cs:18]: Done

File: Exercise_3/MainActivity.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        {
+            tvResultTitle.Visibility = ViewStates.Invisible;
+            tvResult.Text = "";
+            if (edFirstCofficient.Text.Equals("") ||
+                edSecondCofficient.Text.Equals("") ||
+                edThirdCofficient.Text.Equals(""))
+            {
+                Toast.MakeText(this, GetString(Resource.String.InputData), ToastLength.Short).Show();
```
Comment (voquanghoa) [inline Exercise_3/MainActivity.cs]: How about
```C#
Toast.MakeText(this, Resource.String.InputData, ToastLength.Short).Show();
```

File: Exercise_3/MainActivity.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        [InjectView(Resource.Id.edSecondCofficient)] private EditText edSecondCofficient;
+        [InjectView(Resource.Id.edThirdCofficient)] private EditText edThirdCofficient;
+        [InjectView(Resource.Id.tvResultTitle)] private TextView tvResultTitle;
+        [InjectView(Resource.Id.tvResult)] private TextView tvResult;
+        [InjectOnClick(Resource.Id.btnSolve)]
+        private void Solve(object sender, EventArgs e)
+        {
+            tvResultTitle.Visibility = ViewStates.Invisible;
+            tvResult.Text = "";
+            if (edFirstCofficient.Text.Equals("") ||
```
Comment (voquanghoa) [inline Exercise_3/MainActivity.cs]: Change it to `string.IsNullOrEmpty(edFirstCofficient.Text)`...

File: Exercise_3/MainActivity.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                {
+                    var a = Convert.ToDouble(edFirstCofficient.Text);
+                    var b = Convert.ToDouble(edSecondCofficient.Text);
+                    var c = Convert.ToDouble(edThirdCofficient.Text);
+                    tvResultTitle.Visibility = ViewStates.Visible;
+                    tvResult.Text = SolveEquation(a, b, c);
+                }
+                catch (Exception)
+                {
+                    Toast.MakeText(this, GetString(Resource.String.InputFloat), ToastLength.Short).Show();
```
Comment (voquanghoa) [inline Exercise_3/MainActivity.cs]:
```C#
Toast.MakeText(this,Resource.String.InputFloat, ToastLength.Short).Show();
```

File: Exercise_3/MainActivity.cs [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+        private string SolveEquation(double a, double b, double c)
+        {
+            if (a.Equals(0))
+            {
+                return b.Equals(0)
+                    ? (c.Equals(0)
+                        ? GetString(Resource.String.InfiniteSolution)
+                        : GetString(Resource.String.NoSolution))
+                    : $"Phương trình đã cho có một nghiệm:\n\t\tx = {-c / b}";
```
Comment (voquanghoa) [inline Exercise_3/MainActivity.cs]: Move the string to String.xml and use the function String.Format("....",xxx)

File: Exercise_3/MainActivity.cs (hunk 4 above)
Comment (nguyenvanman) [reply Exercise_3/MainActivity.cs]: done

File: Exercise_3/MainActivity.cs (hunk 3 above)
Comment (nguyenvanman) [reply Exercise_3/MainActivity.cs]: Done

File: Exercise_3/MainActivity.cs (hunk 5 above)
Comment (nguyenvanman) [reply Exercise_3/MainActivity.cs]: Done

**Pull request merged**